main.cpp

```diff
@@ -35,39 +30,69 @@ Gnuplot g5("lines");
 void plot_line_gnu(double A, double B, uint16_t size, Gnuplot *g, bool hold_on, string title);
 void plot_gnu(int32_t *array, uint16_t size, Gnuplot *g, bool hold_on, string title);
 
-int main()
+int main(int argc, char *argv[])
 {
-// SELECT WHICH DATABASE YOU WANT:
-//TODO: make this variables for the main function
+
+  //check if too many arguments are passed
+  if (argc != 3) {
+    cout << "Wrong amount of arguments!" << endl;
+    cout << "Usage: ./testing 'stereoboard_nr' 'take_nr'" << endl;
+    cout << "Example: ./testing 1 16" << endl;
+
+    return 0;
+  }
+  cout << "stereoboard " << argv[1] << " with take " << argv[2] << endl;
 
   string configuration_board = "forward_camera";
-  int number_stereoboard = 1;
-  int number_take = 16;	// 3
-  float time_inc = 1/7.; // 1/8
+  int number_stereoboard = atoi(argv[1]);
+  int number_take = atoi(argv[2]);
 
   uint32_t frame_time = 0;
 
   // Find Directories
   stringstream file_directory_images;
+  stringstream file_directory_timing;
   stringstream file_directory_calibration;
   stringstream file_directory_results;
 
   file_directory_images << "stereoboard_database/database_stereoboard_" << number_stereoboard << "/" <<
                         configuration_board << "/take" << number_take << "/%1d.bmp";
+  file_directory_timing << "stereoboard_database/database_stereoboard_" << number_stereoboard << "/" <<
+                        configuration_board << "/take" << number_take << "/timing.dat";
   file_directory_calibration << "stereoboard_database/database_stereoboard_" << number_stereoboard <<
                              "/calibration_data.txt";
   file_directory_results << "stereoboard_database/database_stereoboard_" << number_stereoboard << "/" <<
                          configuration_board << "/take" << number_take << "/result_stereo.csv";
 
-
   //initialize for edgeflow
-  edgeflow_init(128, 94, 0);
+  edgeflow_init(128, 94, 0, NULL);
+
+
+  //Uncomment if you want to try an old version of the code
+  //edgeflow_init(128, 94, 0);
 
   //Open files needed for testing
   VideoCapture cap; cap.open(file_directory_images.str());    //image location
   ofstream output; output.open(file_directory_results.str());   // result file
   fstream calibration_file(file_directory_calibration.str(), ios_base::in);
   calibration_file >> edgeflow_params.stereo_shift; // calibration data of that file
+  fstream timing_file(file_directory_timing.str(), ios_base::in);
+
+  //Make vector with timing data
+  double num;
+  string line, temp_str_c1, temp_str_c2;
+  vector<double> timing;
+  while (getline(timing_file,line))
+  {
+	  istringstream ss(line);
+	  ss>>temp_str_c1; // read first column
+	  ss>>temp_str_c2; // read second column
+	  if(atof(temp_str_c2.c_str())==1.0) // If second column is not 1, then there is no accommodating image
+	  {
+		  num = (double)atof(temp_str_c1.c_str());
+		  timing.push_back(num);
+	  }
+  }
```

Comment:
tab size. if you are using eclipse you can apply the same code style as in paparazzi to this project. In eclipse Project->Properties->C/C++ General->Formatter
select enable project specific settings and select the paparazzi stlye: https://github.com/paparazzi/paparazzi/blob/master/paparazzi_code_profile_eclipse.xml